kalite/topic_tools/__init__.py

```diff
@@ -45,24 +42,38 @@
 
 
 # Globals that can be filled
-TOPICS          = None
+TOPICS          = {}
 CACHE_VARS.append("TOPICS")
-def get_topic_tree(force=False, props=None):
-    global TOPICS, TOPICS_FILEPATH
-    if TOPICS is None or force:
-        TOPICS = softload_json(TOPICS_FILEPATH, logger=logging.debug, raises=False)
-        validate_ancestor_ids(TOPICS)  # make sure ancestor_ids are set properly
-
-        # Limit the memory footprint by unloading particular values
-        if props:
-            node_cache = get_node_cache()
-            for kind, list_by_kind in node_cache.iteritems():
-                for node_list in list_by_kind.values():
-                    for node in node_list:
-                        for att in node.keys():
-                            if att not in props:
-                                del node[att]
```

Comment:
Aww, but this code was so beautiful!!!

![image](https://cloud.githubusercontent.com/assets/618416/5731503/f61d290e-9b38-11e4-903e-0c8f011c9340.png)